File: blocks/archiva/first-login/scripts/first-login.rb
require "selenium-webdriver"

wait = Selenium::WebDriver::Wait.new(:timeout => 5)

driver = Selenium::WebDriver.for :remote, url: ENV['PHANTOMJS_URL']

target_size = Selenium::WebDriver::Dimension.new(1024, 768)
driver.manage.window.size = target_size

user = ENV['SERVICE_ARCHIVA_ESADMIN_CREDENTIALS_USER_NAME']
pass = ENV['SERVICE_ARCHIVA_ESADMIN_CREDENTIALS_PASSWORD']
email = ENV['SERVICE_ARCHIVA_ESADMIN_CREDENTIALS_EMAIL']

# user: assumed to be 'admin' - can't override in Archiva

driver.navigate.to "#{ENV["ARCHIVA_URL"]}"

sleep 5

driver.find_element(:id, "create-admin-link-a").click

sleep 5

driver.find_element(:id, "password").send_keys pass
driver.find_element(:id, "confirmPassword").send_keys pass

driver.find_element(:id, "email").send_keys email

driver.find_element(:id, "validated").click

driver.find_element(:id, "user-create-form-register-button").click

sleep 5

driver.quit

File: blocks/archiva/first-login/scripts/first-login.js
// APP_LOCAL_GITEA_URL

module.exports = {
    'Archiva First Login' : function (browser) {
      console.log("Archiva: " + browser.config.archiva.url)
      browser
        .url(browser.config.archiva.url)
        .waitForElementVisible('#create-admin-link-a', 5000)
        .click("#create-admin-link-a")
        .waitForElementVisible("input[name='password']", 3000)
        .setValue("input[name='password']", process.env.SERVICE_ARCHIVA_ESADMIN_CREDENTIALS_PASSWORD)
        .setValue("input[name='confirmPassword']", process.env.SERVICE_ARCHIVA_ESADMIN_CREDENTIALS_PASSWORD)
        .setValue("input[name='email']", process.env.SERVICE_ARCHIVA_ESADMIN_CREDENTIALS_EMAIL)
        .click("input[name='validated']")
        .waitForElementVisible("#user-create-form-register-button", 3000)
        .click("#user-create-form-register-button")

    }
  };
